gooddata-java-model/src/main/java/com/gooddata/sdk/model/hierarchicalconfig/ConfigItem.java

```diff
@@ -0,0 +1,154 @@
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.model.hierarchicalconfig;
+
+import com.fasterxml.jackson.annotation.*;
+import com.gooddata.sdk.common.util.GoodDataToStringBuilder;
+
+
+import static com.gooddata.sdk.common.util.Validate.*;
+import static com.gooddata.sdk.model.hierarchicalconfig.SourceType.*;
+
+/**
+ * Contains information about hierarchical configuration object aka platform setting aka feature flag.
+ * Purpose of config item is carrying information about some specific feature of Gooddata platform.
+ */
+@JsonTypeName("settingItem")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ConfigItem {
+
+    public static final String CLIENT_CONFIG_ITEM_URI = CLIENT.getApiUri() + "/{configName}";
+    public static final String DATA_PRODUCT_CONFIG_ITEM_URI = DATA_PRODUCT.getApiUri() + "/{configName}";
+    public static final String SEGMENT_CONFIG_ITEM_URI = SEGMENT.getApiUri() + "/{configName}";
+    public static final String DOMAIN_CONFIG_ITEM_URI = DOMAIN.getApiUri() + "/{configName}";
+    public static final String PROJECT_CONFIG_ITEM_URI = PROJECT.getApiUri() + "/{configName}";
+    public static final String PROJECT_GROUP_CONFIG_ITEM_URI = PROJECT_GROUP.getApiUri() + "/{configName}";
+
+    private final String key;
+    private String value;
+
+    @JsonIgnore
+    private final String source;
+    @JsonIgnore
+    private final Links links;
+
+    /**
+     * Creates new config item with given key and value.
+     *
+     * @param key unique name of config item
+     * @param value value of config item
+     */
+    public ConfigItem(String key, String value) {
+        this(notEmpty(key, "key"), notEmpty(value, "value"), null, null);
+    }
+
+    @JsonCreator
+    ConfigItem(@JsonProperty("key") String key, @JsonProperty("value") String value,
+               @JsonProperty("source") String source, @JsonProperty("links") Links links) {
+        this.key = key;
+        this.value = value;
+        this.source = source;
+        this.links = links;
+    }
+
+    @JsonProperty("key")
+    public String getKey() {
+        return key;
+    }
+
+    @JsonIgnore
+    public String getName() { return key; }
+
+    @JsonProperty("value")
+    public String getValue() {
+        return value;
+    }
+
+    @JsonIgnore
+    public String getSource() {
+        return source;
+    }
+
+    public Links getLinks() {
+        return links;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    @JsonIgnore
+    public boolean isEnabled() {
+        return Boolean.parseBoolean(this.value);
+    }
+
+    @JsonIgnore
+    public void setEnabled(Boolean value) {
+        this.value = String.valueOf(value);
+    }
+
+    @JsonIgnore
+    public SourceType getSourceType() {
```

Comment:
I would just add javadoc explaining, that this method can return null in case when source is not recognised by given version of this library.
Also I miss unit tests for this and some other methods (I would test even those synonyms).